internal/scm/bitbucketcloud/bitbucket_cloud.go

```diff
@@ -0,0 +1,324 @@
+	repoPRs, _ := bbc.getPullRequests(ctx, repoSlug[1])
+	for _, repoPR := range repoPRs {
+		pr := pullRequest(repoPR)
+		if pr.branchName == branchName && pr.status == scm.PullRequestStatusSuccess {
+			return repoPR, nil
+		}
+	}
+	return nil, nil
+}
+
+func (bbc *BitbucketCloud) MergePullRequest(ctx context.Context, pr scm.PullRequest) error {
+	bbcPR := pr.(pullRequest)
+	repoSlug := strings.Split(bbcPR.guiURL, "/")
+	prOptions := &bitbucket.PullRequestsOptions{
+		ID:           fmt.Sprintf("%d", bbcPR.number),
+		SourceBranch: bbcPR.branchName,
+		RepoSlug:     repoSlug[4],
+		Owner:        bbc.workspaces[0],
+	}
+	_, err := bbc.bbClient.Repositories.PullRequests.Merge(prOptions)
+	return err
+}
+
+func (bbc *BitbucketCloud) ClosePullRequest(ctx context.Context, pr scm.PullRequest) error {
+	bbcPR := pr.(pullRequest)
+	repoSlug := strings.Split(bbcPR.guiURL, "/")
+	prOptions := &bitbucket.PullRequestsOptions{
+		ID:           fmt.Sprintf("%d", bbcPR.number),
+		SourceBranch: bbcPR.branchName,
+		RepoSlug:     repoSlug[4],
+		Owner:        bbc.workspaces[0],
+	}
+	_, err := bbc.bbClient.Repositories.PullRequests.Decline(prOptions)
+	return err
+}
+
+func (bbc *BitbucketCloud) GetRepositories(ctx context.Context) ([]scm.Repository, error) {
+	repoOptions := &bitbucket.RepositoriesOptions{
+		Role:  "member",
+		Owner: bbc.workspaces[0],
+	}
+
+	repos, err := bbc.bbClient.Repositories.ListForAccount(repoOptions)
+
+	if err != nil {
+		return nil, err
+	}
+
+	repositories := make([]scm.Repository, 0, len(repos.Items))
+	for _, repo := range repos.Items {
+		if slices.Contains(bbc.repositories, repo.Name) {
+			converted, err := bbc.convertRepository(repo)
+			if err != nil {
+				return nil, err
+			}
+			repositories = append(repositories, *converted)
+		}
+	}
+
+	return repositories, nil
+}
+
+func (bbc *BitbucketCloud) ForkRepository(ctx context.Context, repo scm.Repository, newOwner string) (scm.Repository, error) {
+	splitRepoFullName := strings.Split(repo.FullName(), "/")
+
+	if newOwner == "" {
+		newOwner = bbc.username
+	}
+	options := &bitbucket.RepositoryForkOptions{
+		FromOwner: bbc.workspaces[0],
+		FromSlug:  splitRepoFullName[1],
+		Owner:     newOwner,
+		Name:      splitRepoFullName[1],
+	}
+	// TODO: Support for selecting Bitbucket project to fork into
+	resp, err := bbc.bbClient.Repositories.Repository.Fork(options)
+	if err != nil {
+		return nil, err
+	}
+	res, err := bbc.convertRepository(*resp)
+	if err != nil {
+		return nil, err
+	}
+	return *res, nil
+}
+
+func (bbc *BitbucketCloud) convertRepository(repo bitbucket.Repository) (*repository, error) {
+	var cloneURL string
+
+	rLinks := &repoLinks{}
+	linkBytes, _ := json.Marshal(repo.Links)
+	_ = json.Unmarshal(linkBytes, rLinks)
+
+	if bbc.sshAuth {
+		cloneURL = findLinkType(rLinks.Clone, cloneSSHType)
+		if cloneURL == "" {
+			return nil, errors.Errorf("unable to find clone url for repository %s using clone type %s", repo.Name, cloneSSHType)
+		}
```

Comment:
Move the error to inside the `findLinkType` function, and let it return an error